util/comparator.cc

```diff
@@ -220,4 +224,52 @@ const Comparator* ReverseBytewiseComparator() {
   return &rbytewise;
 }
 
+#ifndef ROCKSDB_LITE
+static int RegisterBuiltinComparators(ObjectLibrary& library,
+                                      const std::string& /*arg*/) {
+  library.Register<Comparator>(
+      BytewiseComparatorImpl::kClassName(),
+      [](const std::string& /*uri*/, std::unique_ptr<Comparator>* /*guard */,
+         std::string* /* errmsg */) {
+        return const_cast<Comparator*>(BytewiseComparator());
+      });
+  library.Register<Comparator>(
+      ReverseBytewiseComparatorImpl::kClassName(),
+      [](const std::string& /*uri*/, std::unique_ptr<Comparator>* /*guard */,
+         std::string* /* errmsg */) {
+        return const_cast<Comparator*>(ReverseBytewiseComparator());
+      });
+  return 2;
+}
+#endif  // ROCKSDB_LITE
+
+static bool LoadComparator(const std::string& id, Comparator** result) {
+  bool success = true;
+  if (id == BytewiseComparatorImpl::kClassName()) {
+    *result = const_cast<Comparator*>(BytewiseComparator());
+  } else if (id == ReverseBytewiseComparatorImpl::kClassName()) {
+    *result = const_cast<Comparator*>(ReverseBytewiseComparator());
+  } else {
+    success = false;
+  }
+  return success;
+}
+
+Status Comparator::CreateFromString(const ConfigOptions& config_options,
+                                    const std::string& value,
+                                    const Comparator** result) {
```

Comment:
Yes, it is defined here as const because the option is const.  This actually makes everything else slightly harder to do.  If I make them non-const, it also will break (at least UBSAN) compatibility with any code that currently uses the ObjectRegistry to register their comparators.